src/components/stepper/stepper.ts

```diff
@@ -0,0 +1,122 @@
+  providers: [MdIconRegistry],
+  directives: [MdIcon],
+  encapsulation: ViewEncapsulation.None
+})
+export class MdStepper {
+
+  steps: MdStep[] = [];
+
+  @Input() mode: 'linear' | 'nonlinear' = 'nonlinear';
+
+  addStep(step: MdStep): void {
+    if (this.steps.length === 0) step.active = true;
+    this.steps.push(step);
+  }
+
+  selectStep(step: MdStep): void {
+    this.currentStep.completed = this.currentStep.valid ? true : false;
+    if (this.mode === 'linear') {
+      if (!this.currentStep.valid) return;
+      //Checks if all steps before are completed or optional
+      for (var i = this.steps.indexOf(this.currentStep); i < this.steps.indexOf(step); i++) {
+        if (!this.steps[i].completed && !this.steps[i].optional) return;
+      }
+    }
+    if (step.completed && !step.editable) return;
+    this.steps.forEach((step) => {
+      step.active = false;
+    });
+    step.active = true;
+  }
+
+  get currentStep(): MdStep {
+    return this.steps.filter(step => step.active)[0];
+  }
+
+  selectNextStep(step: MdStep): void {
+    if (this.getNextEditableStep(step)) this.selectStep(this.getNextEditableStep(step));
+  }
+
+  selectPreviousStep(step: MdStep): void {
+    if (this.getPreviousEditableStep(step)) this.selectStep(this.getPreviousEditableStep(step));
+  }
+
+
+  getNextEditableStep(step: MdStep): MdStep {
+    if (this.steps.indexOf(this.currentStep) === this.steps.length - 1) return undefined;
+    for (var i = this.steps.indexOf(this.currentStep) + 1; i < this.steps.length; i++) {
+      if (this.steps[i].isTargetable) {
+        return this.steps[i];
+      }
+    }
+    return undefined;
+  }
+
+  getPreviousEditableStep(step: MdStep): MdStep {
+    if (this.steps.indexOf(this.currentStep) === 0) return undefined;
+    for (var i = this.steps.indexOf(this.currentStep) - 1; i >= 0; i--) {
+      if (this.steps[i].isTargetable) {
+        return this.steps[i];
+      }
+    }
+    return undefined;
+  }
+}
+
+@Component({
+  selector: 'md-step',
+  templateUrl: 'components/stepper/step.html',
+  styleUrls: ['components/stepper/step.css'],
+  encapsulation: ViewEncapsulation.None
+})
+export class MdStep {
+
+  active: boolean = false;
+  completed: boolean = false;
+
+  @Input() @BooleanFieldValue() optional: boolean = false;
+  @Input() @BooleanFieldValue() editable: boolean = true;
+  @Input() @BooleanFieldValue() valid: boolean = true;
+  @Input() label: string = '';
+
+  constructor(private steps: MdStepper) {
+    steps.addStep(this);
+  }
+
+  get isLeft(): boolean {
+    return this.steps.steps.indexOf(this) < this.steps.steps.indexOf(this.steps.currentStep);
+  }
+
+  get isRight(): boolean {
+    return this.steps.steps.indexOf(this) > this.steps.steps.indexOf(this.steps.currentStep);
+  }
+
+  get isTargetable(): boolean {
+    return !this.editable && this.completed ? false : true;
+  }
+
+  private get _stepperIconLigature(): string {
+    return (!this.active && this.completed) ? 'done' : 'create';
```

Comment:
Some users, may not have the material design icon font included. I think, (once) this component will be available, the SVG should be inlined.